internal/route/api/v1/repo/file.go

```diff
@@ -45,6 +48,72 @@ func GetArchive(c *context.APIContext) {
 	repo.Download(c.Context)
 }
 
+func GetRepoTree(c *context.APIContext) {
+	repoPath := db.RepoPath(c.Params(":username"), c.Params(":reponame"))
+	gitRepo, err := git.OpenRepository(repoPath)
+
+	if err != nil {
+		c.ServerError("OpenRepository", err)
+		return
+	}
+	c.Repo.GitRepo = gitRepo
+	gitTree, err := gitRepo.GetTree(c.Params(":sha"))
+	if err != nil {
+		c.NotFoundOrServerError("GetRepoTree", git.IsErrNotExist, err)
+		return
+	}
+	entries, err := gitTree.ListEntries()
+	if err != nil {
+		c.ServerError("GetRepoTree", err)
+		return
+	}
+	if entries == nil {
+		c.NotFound()
```

Comment:
Is it really not found? Or it's just an empty tree/directory.

Reply:
Yea, you're right. It's shouldn't be not found, it should be just an empty tree. I am going to preserve the top level structure but return an empty tree. 

Which will ook like:
```bash
[
  {
    "url":"http://localhost:3000/api/v1/repos/root/testrepo/git/trees/c59441ded1549b149def0d4c54594d31a7f3718f"
    "sha": "c59441ded1549b149def0d4c54594d31a7f3718f",
    "tree": []
}
```